nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -248,6 +220,43 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
 
     }
 
+    protected void writeActors(List<String> actors, boolean isFetchActors, JsonGenerator jg) throws IOException {
+        for (String actorId : actors) {
+            if (isFetchActors) {
+                Object actor = fetchActor(actorId);
+                if (actor != null) {
+                    writeEntity(actor, jg);
+                }
+            } else {
+                jg.writeStartObject();
+                jg.writeStringField("id", actorId);
+                jg.writeEndObject();
+            }
+        }
+    }
```

Comment:
It seems like the behavior has changed: if `isFetchActors` is true and no user or group is found then nothing is written.
With the current code, I believe that the principal's id is written anyway: case where `continue` is never executed in the ` if (isFetchActors) {...}` block.

Reply:
Good point, thanks!
I'm still doing some improvements and trying to add specific unit tests because it was something that I was trying to avoid, changing behaviour / breaking compatibility.